src/domain/flow-system/domain/src/entities/shared/batching_rule.rs

```diff
@@ -8,13 +8,16 @@
 // by the Apache License, Version 2.0.
 
 use chrono::Duration;
+use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[serde_with::serde_as]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
 pub struct BatchingRule {
     min_records_to_await: u64,
+    #[serde_as(as = "serde_with::DurationMicroSeconds<i64>")]
```

Comment:
Why would we need microseconds? In UI we minimally allow minutes!

Reply:
I wanted to make a maximum match to the `Duration` type (nanoseconds inside) -- but, you're right, in terms of our business logic, it's unnecessary -- chose seconds as the minimum unit available

Done

Reply:
The default was microseconds, right?

Reply:
No, nanoseconds:
```rust
#[stable(feature = "duration", since = "1.3.0")]
#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
#[cfg_attr(not(test), rustc_diagnostic_item = "Duration")]
pub struct Duration {
    secs: u64,
    nanos: Nanoseconds, // Always 0 <= nanos < NANOS_PER_SEC
}
```